src/nvim/api/extmark.c

```diff
@@ -1038,15 +1038,17 @@ void nvim_buf_clear_namespace(Buffer buffer, Integer ns_id, Integer line_start,
 ///                 ["win", winid, bufnr, row]
 ///             - on_end: called at the end of a redraw cycle
 ///                 ["end", tick]
+///             - redraw (boolean): redraw immediately after this function is called (default true)
 void nvim_set_decoration_provider(Integer ns_id, Dict(set_decoration_provider) *opts, Error *err)
   FUNC_API_SINCE(7) FUNC_API_LUA_ONLY
 {
   DecorProvider *p = get_decor_provider((NS)ns_id, true);
   assert(p != NULL);
   decor_provider_clear(p);
 
-  // regardless of what happens, it seems good idea to redraw
-  redraw_all_later(UPD_NOT_VALID);  // TODO(bfredl): too soon?
+  if (GET_BOOL_OR_TRUE(opts, set_decoration_provider, redraw)) {
+    redraw_all_later(UPD_NOT_VALID);
+  }
```

Comment:
What would be the cons of just never redrawing?

Reply:
Alternatively, could we just avoid the redraw if we know the intro screen is showing?